### PR TITLE
Install CIRCT out of Conda

### PR DESCRIPTION
**Pull request by abejgonzalez**
The CIRCT community has some scripts to help install updated versions / update versions of CIRCT in repos. This PR is to see how well this works in Chipyard. There are two potential implementations of using these scripts:

1. This PR. I.e. handle CIRCT installation/updating separate from Conda. Allows us to remove the conda-feedstock for CIRCT and have less things inside of Conda (environment bloat).
2. Update the conda-feedstock to automatically update and publish using these scripts. I.e. the conda-feedstock will download CIRCT, put it in the conda location, and publish it. This will keep the same type of flow as before (all in conda) and will just require us to update firtool ourselves (maybe we can write a script to autoupdate it).

EDIT: We decided on #1 since it's shifts the burden of us having to maintain a feedstock to the CIRCT devs who already publish a working version of CIRCT.

Any thoughts?

Side Update: Moves the `env.sh` generation into the conda setup part of `build-setup.sh`. Thus if a later part of the flow fails you still have a `env.sh` to diagnose and manually do other parts of the setup.

**Related PRs / Issues**:

**Type of change**:
- [ ] Bug fix
- [ ] New feature
- [ ] Other enhancement

**Impact**:
- [ ] RTL change
- [ ] Software change (RISC-V software)
- [ ] Build system change
- [ ] Other

**Contributor Checklist**:
- [ ] Did you set `main` as the base branch?
- [ ] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [ ] Did you state the type-of-change/impact?
- [ ] Did you delete any extraneous prints/debugging code?
- [ ] Did you mark the PR with a `changelog:` label?
- [ ] (If applicable) Did you update the conda `.conda-lock.yml` file if you updated the conda requirements file?
- [ ] (If applicable) Did you add documentation for the feature?
- [ ] (If applicable) Did you add a test demonstrating the PR?

- [ ] (If applicable) Did you mark the PR as `Please Backport`?

### CODE REVIEW

**Comment by seldridge** [comment]
I'm glad to see these GitHub actions getting some use. I'm also glad that the factoring of `circt/install-circt` to be predominantly shell-based is working for you. If I ever find some time, I plan to convert `circt/update-circt` to also be similarly shell-based.

These things should be usable by flows which aren't GitHub Actions, i.e.

File: .github/actions/create-conda-env/action.yml [hunk 1]
```diff
@@ -21,5 +21,13 @@ runs:
...
           ./scripts/build-toolchain-extra.sh esp-tools -p $CONDA_PREFIX/esp-tools
           conda deactivate
+
+          echo "Installing CIRCT to conda area"
+          $REMOTE_CHIPYARD_DIR/conda-reqs/install-circt/bin/download-release-or-nightly-circt.sh \
+              -f circt-full-shared-linux-x64.tar.gz \
+              -i $CONDA_PREFIX \
+              -v version-file \
+              -x ./conda-reqs/circt.json \
+              -g null
```
[the remark this answers is not in the document]
Comment (seldridge) [reply .github/actions/create-conda-env/action.yml:32]: This should be fine, but you will get rate limited to 60 requests per hour without the token. GitHub might waive this if it originates from a GitHub runner, but I'm not sure. (The token is part of the API request made to GitHub and GitHub seems to accept dummy or empty tokens: https://github.com/circt/install-circt/blob/main/bin/download-release-or-nightly-circt.sh#L77)

[the remark this answers is not in the document]
Comment (abejgonzalez) [reply .github/actions/create-conda-env/action.yml:32]: Ah. I see. That should be easy enough to add in the CI. For our normal setup script not having a token seems fine (I couldn't find if there is some default token for the git CLI that I could use).

File: scripts/build-setup.sh [hunk 2]
```diff
@@ -148,10 +148,39 @@ if run_step "1"; then
...
+    echo "::ERROR:: you must have conda in your environment first"
+    return 1  # don't want to exit here because this file is sourced
+fi
+
+# if we're sourcing this in a sub process that has conda in the PATH but not as a function, init it again
+conda activate --help >& /dev/null || source $(conda info --base)/etc/profile.d/conda.sh
+\0
+END_CONDA_ACTIVATE
+
+    replace_content env.sh build-setup-conda "# line auto-generated by $0
```
Comment (jerryz123) [inline scripts/build-setup.sh:180]: Spacing bug?

**Review by jerryz123 — CHANGES_REQUESTED** [review]
Since the CIRCT stuff is not conda-specific, I think it can be outside of the conda directory.

File: scripts/build-setup.sh (hunk 2 above)
Comment (abejgonzalez) [reply scripts/build-setup.sh:180]: There shouldn't be (assuming you are referring to `$CONDA_ACTIVATE_PREAMBLE` being at the start of the line)

Comment (jerryz123) [inline scripts/build-setup.sh:180]:
```suggestion
replace_content env.sh build-setup-conda "# line auto-generated by $0
```
?

**Review by jerryz123 — APPROVED** [review]

**Pull request merged**